Bring supported Ruby versions up to speed

etagwerker opened the pull request:
Trying to fix Travis CI builds.

etagwerker committed:
Bring supported Ruby versions up to speed

bronzdoc commented on .travis.yml:
```diff
@@ -9,11 +9,10 @@ before_install:
-  - 2.0.0
-  - 1.9.2
-  - 1.9.3
+  - 2.3
```
2.3 is not supported anymore, i'll personally not support this ruby

etagwerker committed:
Relax `guard` and `guard-bundler` dependencies

Right now all builds are failing with this error: https://travis-ci.org/fastruby/metric_fu/jobs/584614366

etagwerker committed:
Remove ruby-2.3 from supported list

While 2.3 seems to work fine, I don't think we should maintain it moving forward. It has been EOLed.

etagwerker committed:
Allow `jruby` to fail in Travis CI
